src/graphics/program-lib/phong.js

```diff
@@ -746,6 +767,42 @@ pc.programlib.phong = {
         code += "\n";
         code += getSnippet(device, 'common_main_end');
 
+        if (hasPointLights) {
+            code = chunks.lightDirPointPS + code;
+        }
+        if (usesLinearFalloff) {
+            code = chunks.falloffLinearPS + code;
+        }
+        if (usesInvSquaredFalloff) {
+            code = chunks.falloffInvSquaredPS + code;
+        }
+        if (usesSpot) {
+            code = chunks.spotPS + code;
+        }
+        var structCode = "struct psInternalData {\n";
+        if (code.includes("data.reflection")) structCode += "vec4 reflection;\n";

```

Comment:
I just checked MDN. Seems String.includes is not universally supported. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/includes